src/commands/EditDataSourceEntryCommand.ts

```diff
@@ -14,46 +14,115 @@
+import { DataVirtConfig, DataSourceConfig, Property, ValueFrom, SecretRef, ConfigMapRef, KeyRef } from '../model/DataVirtModel';
 
-export function editDataSourceEntryCommand(ctx) {
-	const item: DataSourceConfigEntryTreeNode = ctx;
-	vscode.window.showInputBox( {value: item.getValue()})
+export function editDataSourceEntryCommand(dsEntryTreeNode: DataSourceEntryTreeNode) {
+	const dsConfig: DataSourceConfig = dsEntryTreeNode.getParent().dataSourceConfig;
+	const entry: Property = utils.getDataSourceEntryByName(dsEntryTreeNode.getKey(), dsConfig);
+	if (entry.valueFrom) {
+		editReferenceEntryType(dsEntryTreeNode, dsConfig, entry);
+	} else {
+		editValueEntryType(dsEntryTreeNode, dsConfig, entry);
+	}
+}
+
+export function editValueEntryType(dsEntryTreeNode: DataSourceEntryTreeNode, dsConfig: DataSourceConfig, entry: Property) {
+	vscode.window.showInputBox( { prompt: 'Enter a new value', placeHolder: `${entry.value ? entry : '<empty>'}`, value: entry.value })
 		.then( ( newValue: string) => {
-			handleDataSourceEntryEdit(item.getProject().dvConfig, item.getParent().dsConfig, item.getProject().getFile(), item.getKey(), newValue)
+			if (!newValue) {
+				return;
+			}
+			handleDataSourceEntryEdit(dsEntryTreeNode.getProject().dvConfig, dsConfig, dsEntryTreeNode.getProject().getFile(), dsEntryTreeNode.getKey(), newValue)
 				.then( (success: boolean) => {
-					if (success) {
-						vscode.window.showInformationMessage(`DataSource entry has been modified...`);
+					showFeedback(success, dsEntryTreeNode.getKey(), dsConfig.name);
+				});
+		});
+}
+
+export function editReferenceEntryType(dsEntryTreeNode: DataSourceEntryTreeNode, dsConfig: DataSourceConfig, entry: Property) {
+	const ref: ValueFrom = entry.valueFrom;
+	let oldName: string;
+	let oldKey: string;
+	if (utils.isSecretRef(ref.valueFrom)) {
+		const secRef : SecretRef = ref.valueFrom;
+		oldName = secRef.secretKeyRef.name;
+		oldKey = secRef.secretKeyRef.key;
+	} else if (utils.isConfigMapRef(ref.valueFrom)) {
+		const mapRef : ConfigMapRef = ref.valueFrom;
+		oldName = mapRef.configMapKeyRef.name;
+		oldKey = mapRef.configMapKeyRef.key;
+	} else {
+		return;
+	}
+	vscode.window.showInputBox( {prompt: 'Enter the new reference name', value: oldName})
+		.then( (newRefName: string) => {
+			if (!newRefName) {
+				return;
+			}
+			vscode.window.showInputBox( {prompt: 'Enter the new reference key', value: oldKey})
+				.then( (newRefKey: string) => {
+					if (!newRefKey) {
+						return;
+					}
+					let newRef: ValueFrom;
+					let newRefValue: ConfigMapRef | SecretRef;
+					if (utils.isSecretRef(ref.valueFrom)) {
+						newRefValue = new SecretRef(new KeyRef(newRefName, newRefKey));
+					} else if (utils.isConfigMapRef(ref.valueFrom)) {
+						newRefValue = new ConfigMapRef(new KeyRef(newRefName, newRefKey));
 					} else {
-						vscode.window.showErrorMessage(`An error occured when trying to modify the datasource entry...`);
+						return;
```

Comment:
Why no error here? Are we looking for a third option?

Reply:
Good point. So there are only 2 types of references allowed atm which are Secrets and ConfigMaps. If the user edits a reference item it has to be one of these two. So the third case will actually never be hit and is just a dummy. What do you think would be the better solution?

Reply:
I would at least log that something weird happened so the user has some idea. I can't imagine that will ever be the case, but good to catch it